feeadjuster/feeadjuster.py

```diff
@@ -13,45 +13,51 @@
 plugin.update_threshold = 0.05
 
 
-def get_ratio(our_percentage):
+def get_ratio(plugin: Plugin, our_percentage):
     """
     Basic algorithm: the farther we are from the optimal case, the more we
     bump/lower.
     """
-    return 50**(0.5 - our_percentage)
+    return plugin.ratio_base**(0.5 - our_percentage)
 
 
 def maybe_adjust_fees(plugin: Plugin, scids: list):
+    channels_adjusted = 0
     for scid in scids:
         our = plugin.adj_balances[scid]["our"]
         total = plugin.adj_balances[scid]["total"]
         percentage = our / total
         last_percentage = plugin.adj_balances[scid].get("last_percentage")
 
-        # reset to normal fees if imbalance is not high enough
-        if (percentage > plugin.imbalance and percentage < 1 - plugin.imbalance):
-            plugin.rpc.setchannelfee(scid)  # applies default values
-            plugin.log("Set default fees as imbalance is too low: {}".format(scid))
-            return
-
         # Only update on substantial balance moves to avoid flooding, and add
         # some pseudo-randomness to avoid too easy channel balance probing
         update_threshold = plugin.update_threshold
         if not plugin.deactivate_fuzz:
             update_threshold += random.uniform(-0.015, 0.015)
+        if (last_percentage is not None
+                and abs(last_percentage - percentage) < update_threshold):
+            continue
 
-        if (last_percentage is None
-                or abs(last_percentage - percentage) > update_threshold):
-            ratio = get_ratio(percentage)
-            try:
-                plugin.rpc.setchannelfee(scid, int(plugin.adj_basefee * ratio),
-                                         int(plugin.adj_ppmfee * ratio))
-                plugin.log("Adjusted fees of {} with a ratio of {}"
-                           .format(scid, ratio))
-                plugin.adj_balances[scid]["last_percentage"] = percentage
-            except RpcError as e:
-                plugin.log("Could not adjust fees for channel {}: '{}'"
-                           .format(scid, e), level="warn")
+        # reset to normal fees if imbalance is not high enough
+        if (percentage > plugin.imbalance and percentage < 1 - plugin.imbalance):
+            plugin.rpc.setchannelfee(scid)  # applies default values
+            plugin.log("Set default fees as imbalance is too low: {}".format(scid))
+            plugin.adj_balances[scid]["last_percentage"] = percentage
+            channels_adjusted += 1
+            continue
```

Comment:
I intentionally moved it here. I'd like to see this early continue first:
>         if (last_percentage is not None
>                 and abs(last_percentage - percentage) < update_threshold):
>             continue
I have a use-case: I call `forcefeeadjust` after every successful payment. I'd like to avoid multiple call of `setchannelfee`, even if I call `forcefeeadjust` multiple times.
I'll figure out something here.

Reply:
If it was intentional, I'm fine :D

Reply:
Oh, I think I understand now: I broke your test, right? I'll fix that soon.